index.js

```diff
@@ -1,18 +1,80 @@
-const readline = require('readline');
+//import the readline library for cleaner output in the terminal
+const readline = require("readline");
 const rl = readline.createInterface(process.stdin, process.stdout);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
+//random number generator
+function randNum(min, max) {
+  let range = max - min + 1;
+  return Math.floor(Math.random() * range) + min;
+}
 
-start();
-
+// have the computer greet the user and ask if they would like to play
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  console.log(
+    "Let's play a game where you (human) make up a number and I (computer) try to guess it. \n\nFirst let's set the range"
+  );
+  // the user sets the range they want to play with
+  let min = 1;
+  let max = await ask("\nWhat would you like the high range to be? "); // I can't think of a user friendly way to ask this!!!
+  let secretNumber = await ask(
+    "\nWhat would you like your number to be?\nI won't peek, I promise...\n"
+  );
+  console.log("You entered: " + secretNumber);
+  secretNumber = parseInt(secretNumber);
+  // the game!
+
+  let number = min + Math.floor((max - min) / 2);
+  let answer = await ask(`Is your number ${number}? `);
+  let highLow;
+  // if the secretNumber the player entered at the start is not equal to the computer's guess we are in this while loop
+  while (secretNumber !== number) {
+    console.log(number);
+    console.log(secretNumber);
+    highLow = await ask(
+      `Is your number higher(h) or lower(l) than ${number}? `
+    );
+    // if the player says their number was higher than the computer's guess we are here
+    if (highLow === "l") {
+      max = number - 1;
+
+      number = min + Math.floor((max - min) / 2);
+
+      answer = await ask(`Is your number ${number}? `);
+      // if the player says their number was lower than the computer's guess we are here
+    } else if (highLow === "h" && number > min) {
+      min = number + 1;
+      number = min + Math.floor((max - min) / 2);
+
+      answer = await ask(`Is your number ${number}? `);
+    
+  } else if (highLow === "h" && number < min) {
```

Comment:
your cheat detector story is only checking for values when the user is saying their number is higher